fe/fe-core/src/main/java/com/starrocks/planner/PlanNode.java

```diff
@@ -803,6 +803,15 @@ public boolean pushDownRuntimeFilters(DescriptorTable descTbl, RuntimeFilterDesc
 
         // theoretically runtime filter can be applied on multiple child nodes.
         boolean accept = false;
+        List<PlanNode> children = Lists.newArrayList(this.children);
+        if (candidatePartitionByExprs.isEmpty() && this instanceof JoinNode) {
+            JoinNode joinNode = (JoinNode) this;
+            if (joinNode.ukfkProperty != null && joinNode.ukfkProperty.isLeftUK) {
+                // For UK FK join with the fk table as right table(larger table), we should push down runtime filter to
+                // the right table first, because the right table is larger and the runtime filter can filter more data.
```

Comment:
The comment should be:

```
we should reorder the join tables and push down runtime filter to the FK tables.
```
runtime filter is pushed down from right to left

Reply:
this optimization has no effect, so I removed it.